Compiler/BackEnd/Tearing.mo

```diff
@@ -2911,8 +2911,15 @@ algorithm
   if debug then execStat("Tearing.ModifiedCellierHeuristic_3 - 2"); end if;
 
   // 2. Determine the variables in causEq
-  msel := Array.select(mIn,causEq);
-  potentialTVars := List.unique(List.flatten(arrayList(msel)));
+  //msel := Array.select(mIn,causEq);
+
+  potentialTVars := {};
+  for e in causEq loop
+    potentialTVars := listAppend(arrayGet(mIn, e), potentialTVars);
+  end for;
+  potentialTVars := List.unique(potentialTVars);
```

Comment:
@casella If there is a trade off between speed and memory, I would also go for speed, but it seems the change to AvlSetInt increases the memory consumption disproportionally.
I suggest applying this overall improvement first and later we can improve speed if necessary - we should profile the code to see if it really is a bottle neck.